fitbenchmarking/core/fitting_benchmarking.py

```diff
@@ -33,661 +33,544 @@
-    :param data_dir: full path of a directory that holds a group of problem
-                     definition files
-    :type date_dir: str
-    :param checkpointer: The object to use to save results as they're generated
-    :type checkpointer: Checkpoint
-    :param label: The name for the dataset in the checkpoint
-    :type label: str
-
-    :return: all results,
-             problems where all fitting failed,
-             minimizers that were unselected due to algorithm_type
-    :rtype: list[fibenchmarking.utils.fitbm_result.FittingResult],
-            list[str],
-            dict[str, list[str]]
+    The class that performs the fit benchmark
+    and collates the results.
     """
 
-    # Extract problem definitions
-    problem_group = misc.get_problem_files(data_dir)
+    def __init__(self, options, data_dir, checkpointer, label='benchmark'):
+        """
+        Initializes the Fit(ting) class.
+
+        :param options: dictionary containing software used in fitting
+                        the problem, list of minimizers and location of
+                        json file contain minimizers
+        :type options: fitbenchmarking.utils.options.Options
+        :param data_dir: full path of a directory that holds a group of problem
+                         definition files
+        :type date_dir: str
+        :param checkpointer: The object to use to save results as they're
+                             generated
+        :type checkpointer: Checkpoint
+        :param label: The name for the dataset in the checkpoint
+        :type label: str
+        """
+        self._options = options
+        self._data_dir = data_dir
+        self._checkpointer = checkpointer
+        self._label = label
+        self._results = []
+        self._failed_problems = []
+        self._unselected_minimizers = {}
+        self.__start_values_index = 0
+        self.__grabbed_output = output_grabber.OutputGrabber(self._options)
+        self.__emissions_tracker = EmissionsTracker() \
+            if 'emissions' in options.table_type else None
+
+    def benchmark(self):
+        """
+        Call benchmarking on user input and list of paths.
+        The benchmarking structure is:
+
+        .. code-block:: python
+
+            __loop_over_starting_values()
+                __loop_over_software()
+                    __loop_over_minimizers()
+                        __loop_over_jacobians()
+                            __loop_over_hessians()
+                                __perform_fit()
+
+        :return: all results,
+                 problems where all fitting failed, and
+                 minimizers that were unselected due to algorithm_type
+        :rtype: list[fibenchmarking.utils.fitbm_result.FittingResult],
+                list[str],
+                dict[str, list[str]]
+        """
+        problem_group = misc.get_problem_files(self._data_dir)
+
+        problems, results, name_count,  = [], [], {}
```

Comment:
I think we don't need to define results here, as afterwards we overwrite it (line 133)

```suggestion
        problems, name_count = [], {}
```